internal/components/receivers/prometheus.go

```diff
@@ -0,0 +1,149 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package receivers
+
+import (
+	"github.com/go-logr/logr"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
+)
+
+type namespaces struct {
+	Names []string `mapstructure:"names"`
+}
+
+type kubernetesSDConfig struct {
+	Namespaces namespaces `mapstructure:"namespaces"`
+	Role       string     `mapstructure:"role"`
+}
+
+type scrapeConfig struct {
+	KubernetesSDConfigs []kubernetesSDConfig `mapstructure:"kubernetes_sd_configs"`
+	JobName             string               `mapstructure:"job_name"`
+}
+
+type prometheusConfig struct {
+	ScrapeConfigs *[]scrapeConfig `mapstructure:"scrape_configs"`
```

Comment:
does this need to be a pointer to a slice? an empty slice is nil already?

Reply:
Sorry because I made this changes some time ago and I don't remember all the details. But I know there was something that didn't allow me to not use a pointer. Like an error while parsing unset `scrape_configs` or something.

Reply:
ah you probably need to set `omitempty` as an option on mapstructure?